packages/react/src/Details/__tests__/Details.test.tsx

```diff
@@ -102,4 +102,65 @@ describe('Details', () => {
 
     expect(getByTestId('summary')).toHaveTextContent('Open')
   })
+
+  it('Adds default summary if no summary supplied', async () => {
+    const {getByText} = render(<Details data-testid="details">content</Details>)
+
+    expect(getByText('See Details')).toBeInTheDocument()
+    expect(getByText('See Details').tagName).toBe('SUMMARY')
+  })
+
+  it('Adds summary element when supplied as prop', async () => {
+    const {getByText} = render(
+      <Details summary="test summary" data-testid="details">
+        content
+      </Details>,
+    )
+
+    expect(getByText('test summary')).toBeInTheDocument()
+    expect(getByText('test summary').tagName).toBe('SUMMARY')
+  })
+
+  it('Ignores summary prop when child exists', async () => {
+    const {queryByText, getByText} = render(
+      <Details summary="test summary" data-testid="details">
+        <Details.Summary>custom summary</Details.Summary>
+        content
+      </Details>,
```

Comment:
(non blocker) Should this also be a dev warning? [example of warning](https://github.com/primer/react/blob/main/packages/react/src/TooltipV2/Tooltip.tsx#L262-L265)

Reply:
superseeded by https://github.com/primer/react/pull/5015/commits/bac525aa4dd50e934ddaac1c5d7e7dcbf6392874